integrations/google_vertex/tests/test_gemini.py

```diff
@@ -0,0 +1,212 @@
+from unittest.mock import MagicMock, Mock, patch
+
+from haystack.dataclasses import StreamingChunk
+from vertexai.preview.generative_models import (
+    FunctionDeclaration,
+    GenerationConfig,
+    HarmBlockThreshold,
+    HarmCategory,
+    Tool,
+)
+
+from haystack_integrations.components.generators.google_vertex import VertexAIGeminiGenerator
+
+GET_CURRENT_WEATHER_FUNC = FunctionDeclaration(
+    name="get_current_weather",
+    description="Get the current weather in a given location",
+    parameters={
+        "type_": "OBJECT",
+        "properties": {
+            "location": {"type_": "STRING", "description": "The city and state, e.g. San Francisco, CA"},
+            "unit": {
+                "type_": "STRING",
+                "enum": [
+                    "celsius",
+                    "fahrenheit",
+                ],
+            },
+        },
+        "required": ["location"],
+    },
+)
+
+
+@patch("haystack_integrations.components.generators.google_vertex.gemini.vertexai.init")
+@patch("haystack_integrations.components.generators.google_vertex.gemini.GenerativeModel")
+def test_init(mock_vertexai_init, _mock_generative_model):
+
+    generation_config = GenerationConfig(
+        candidate_count=1,
+        stop_sequences=["stop"],
+        max_output_tokens=10,
+        temperature=0.5,
+        top_p=0.5,
+        top_k=0.5,
+    )
+    safety_settings = {HarmCategory.HARM_CATEGORY_DANGEROUS_CONTENT: HarmBlockThreshold.BLOCK_ONLY_HIGH}
+
+    tool = Tool(function_declarations=[GET_CURRENT_WEATHER_FUNC])
+
+    gemini = VertexAIGeminiGenerator(
+        project_id="TestID123",
+        location="TestLocation",
+        generation_config=generation_config,
+        safety_settings=safety_settings,
+        tools=[tool],
+    )
+    mock_vertexai_init.assert_called()
+    assert gemini._model_name == "gemini-pro-vision"
+    assert gemini._generation_config == generation_config
+    assert gemini._safety_settings == safety_settings
+    assert gemini._tools == [tool]
+
+
+@patch("haystack_integrations.components.generators.google_vertex.gemini.vertexai")
```

Comment:
Let’s only mock the init. More mocking shouldn’t be necessary? In case, more mocking is necessary then let’s enumerate it explicitly. Otherwise:
`@patch("haystack_integrations.components.generators.google_vertex.gemini.vertexai")`
->
`@patch("haystack_integrations.components.generators.google_vertex.gemini.vertexai.init")`